src/firestore/firestore_adapter.py

```
@@ -387,4 +409,112 @@ def get_rp_history(self, db, userId, rp_data_limit):
+        valid_info = [info for info in info_list if info['expiration_date'] > now]
+        
+        # 期限切れのデータが存在する場合、データベースを更新
+        if len(valid_info) < len(info_list):
+            doc_ref.update({
+                'info_list': valid_info
+            })
+
+        # ベクトル検索が指定された場合
+        if query_vector is not None:
+            # クエリベクトルをNumPy配列に変換
+            query_array = np.array(query_vector)
```

Ruff (0.8.2)

Undefined name `np`

(F821)